Add new update_fields parameter to update workflow API (#7632)
opensearch-project/documentation-website: _automating-configurations/api/create-workflow.md

Once a workflow is created, provide its `workflow_id` to other APIs.

> The `POST` method creates a new workflow. The `PUT` method updates an existing workflow. With the `update_fields` parameter, only the included fields are updated.

Reviewer: ```suggestion
The `POST` method creates a new workflow. The `PUT` method updates an existing workflow. You can specify the `update_fields` parameter to update specific fields.
```

By default, workflows are validated when they are created to ensure that the syntax is valid and that the graph does not contain cycles. This behavior can be controlled with the `validation` query parameter. If `validation` is set to `all`, OpenSearch performs a complete template validation. Any other value of the `validation` parameter suppresses validation, allowing an incomplete/work-in-progress template to be saved. To disable template validation, set `validation` to `none`:

> You can not update a full workflow that has been provisioned, but you can update fields other than the `workflows` field, such as the name and description.

Reviewer: ```suggestion
You cannot update a full workflow once it has been provisioned, but you can update fields other than the `workflows` field, such as `name` and `description`.
```

```json

> PUT /_plugins/_flow_framework/workflow/<workflow_id>?update_fields=true

Reviewer: Is this a complete request? Do you need a request body that specifies which fields are to be updated?

```
{% include copy-curl.html %}

> You can not use both the `provision` and `update_fields` parameters at the same time.

Reviewer: ```suggestion
You cannot specify both the `provision` and `update_fields` parameters at the same time.
{: .note}
```

> ```json

Reviewer: Please move this up to line 56 (after the description of this request).

```
{% include copy-curl.html %}

> You cannot update a full workflow once it has been provisioned, but you can update fields other than the `workflows` field, such as `name` and `description`.

Reviewer 2: ```suggestion
You cannot update a full workflow once it has been provisioned, but you can update fields other than the `workflows` field, such as `name` and `description`:
```
